synapse/app/_base.py

```diff
@@ -468,16 +467,12 @@ def run_sighup(*args: Any, **kwargs: Any) -> None:
     # everything currently allocated are things that will be used for the
     # rest of time. Doing so means less work each GC (hopefully).
     #
-    # This only works on Python 3.7
-    if platform.python_implementation() == "CPython" and sys.version_info >= (3, 7):
```

Comment:
I think we need to still check the platform implementation is `CPython` here -- I think pypy doesn't have these APIs.

Reply:
pypy has `gc.collect` but not `gc.freeze`

In the interest of duck typing, we should probably check `hasattr(gc, 'freeze')` instead of examining `python_implementation()`

(Or we could say we're standardizing on CPython and rip the Pypy bits out of our codebase; no shame in that)

Reply:
Either of those seems ok, just a shame to delete something that's working.

Reply:
Great, I will re-implement the relevant parts of this check. Since this is already merged, what would be the best way of going about that? Opening a new PR restoring that functionality?

Reply:
> Great, I will re-implement the relevant parts of this check. Since this is already merged, what would be the best way of going about that? Opening a new PR restoring that functionality?

Exactly! It probably makes sense to just use the same changelog entry so they'll get merged by towncrier.